tests/testthat/test_condi.R

```diff
@@ -0,0 +1,71 @@
+expect_condi <- function(x) {
+    checkmate::expect_names(names(x), identical.to = c("x", "message", "correct", "type"))
+    checkmate::expect_character(x$message, null.ok = TRUE)
+    checkmate::expect_logical(x$correct, null.ok = FALSE, len = 1)
+    checkmate::expect_choice(x$type, choices = c("formula", "function", "value"))
+    checkmate::expect_class(x, "grader_condition")
+}
+
+expect_condi_correct <- function(x, message = NULL) {
+    expect_condi(x)
+    expect_equal(x$message, message)
+    expect_true(x$correct)
+}
+
+expect_condi_error <- function(x, message = NULL) {
+    expect_condi(x)
+    expect_equal(x$message, message)
+    expect_false(x$correct)
+}
+
+grader_args <- list()
+learnr_args <- list(last_value = quote(5), envir_prep = new.env())
+condi_formula_t <- condition(~ .result == 5, message = "my correct message", correct = TRUE)
+condi_formula_f <- condition(~ .result == 1, message = "my error message", correct = FALSE)
+
+context("Check condi")
```

Comment:
this should be at the top of the setup (so if an error occurs we can see what context it is in)